### PR TITLE
Prometheus and Grafana pods namespace fix

### PR DESCRIPTION
**Pull request by ashleywang1**
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/1219
resolves https://github.com/solo-io/solo-projects/issues/1146

### CODE REVIEW

File: test/helpers/kube_dump.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ var dumpCommands = func(namespace string) []string {
 	return []string{
 		fmt.Sprintf("echo PODS FROM %s: && kubectl get pod -n %s --no-headers -o custom-columns=:metadata.name", namespace, namespace),
 		fmt.Sprintf("for i in $(kubectl get pod -n %s --no-headers -o custom-columns=:metadata.name); do echo STATUS FOR %s.$i: $(kubectl get pod -n %s $i -o go-template=\"{{range .status.containerStatuses}}{{.state}}{{end}}\"); done", namespace, namespace, namespace),
-		fmt.Sprintf("for i in $(kubectl get pod -n %s --no-headers -o custom-columns=:metadata.name); do echo LOGS FROM %s.$i: $(kubectl logs -n %s $i --all-containers); done", namespace, namespace, namespace),
+		fmt.Sprintf("for i in $(kubectl get pod -n %s --no-headers -o custom-columns=:metadata.name); do echo LOGS FROM %s.$i: $(kubectl logs -n %s $i --all-containers) | grep {\"level\":\"error\" ; done", namespace, namespace, namespace),
```
Comment (kdorosh) [inline test/helpers/kube_dump.go]: why don't we want the full logs from each container? i imagine they can't be too long as the pods were just installed in a very minimalistic fashion.

at the very least, if we're going to filter here, we should use the `-C <num_lines>` `grep` flag to preserve some context around the error.

but i still think we should dump the whole thing

Comment (EItanya) [reply test/helpers/kube_dump.go]: This is kind of a non-issue as that change was part of a different PR. Definitely willing to revisit if those logs were providing more value than we realized

File: projects/gloo/cli/pkg/cmd/install/installer.go [hunk 2]
```diff
@@ -35,6 +37,23 @@ func (i *DefaultGlooKubeInstallClient) WaitForCrdsToBeRegistered(ctx context.Con
 	return waitForCrdsToBeRegistered(ctx, crds)
 }
 
+type NamespacedGlooKubeInstallClient struct {
+	Namespace string
+	Delegate  GlooKubeInstallClient
+	Executor  func(stdin io.Reader, args ...string) error
```
Comment (EItanya) [inline projects/gloo/cli/pkg/cmd/install/installer.go]: I like this, avoid the complexity of the interface

**Pull request merged**